content/en/docs/concepts/workloads/controllers/deployment.md

```diff
@@ -45,6 +45,16 @@ The following is an example of a Deployment. It creates a ReplicaSet to bring up
 In this example:
 
 * A Deployment named `nginx-deployment` is created, indicated by the `.metadata.name` field.
+
+  {{< note >}}
+  When the control plane creates new Pods for a Deployment, the `.metadata.name` of the
+  Deployment is part of the basis for naming those Pods.  For a Deployment, the `.metadata.name`
+  field allows a [DNS subdomain](/docs/concepts/overview/working-with-objects/names#dns-subdomain-names)
+  value, but this can produce unexpected results for the Pod hostnames.  The Kubernetes project recommends
+  that you create Deployments where the `.metadata.name` conforms to the more restrictive rules for a
+  [DNS label](/docs/concepts/overview/working-with-objects/names#dns-label-names).
+  {{< /note >}}
```

Comment:
I wouldn't mention that here. Maybe a brief note and a page-internal link to `#writing-a-deployment-spec`?

If you do that, you could put a similar brief note and another link near:
```markdown
Notice that the name of the ReplicaSet is always formatted as `[DEPLOYMENT-NAME]-[HASH]`.
```

No need to use a `{{< note >}}` for those brief notes.

Reply:
ACK